cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/AbstractConstraint.java

```diff
@@ -24,13 +24,15 @@
 /**
  * A constraint that is checked before executing a schedule.
  */
-public class ConstraintChecker {
+public abstract class AbstractConstraint implements Constraint {
```

Comment:
Why isn't it sufficient that this `AbstractConstraint` acts as the parent of all constraint implementation classes?